src/olympia/addons/models.py

```diff
@@ -541,6 +546,32 @@ def from_upload(cls, upload, platforms, source=None, is_listed=True,
 
         return addon
 
+    @classmethod
+    def resolve_webext_translations(cls, data, upload):
+        """Resolve all possible translations from an add-on.
+
+        This returns a modified `data` dictionary accordingly with proper
+        translations filled in.
+        """
+        if not data.get('is_webextension') or not data.get('default_locale'):
+            # Don't change anything if we don't meet the requirements
+            return data
+
+        fields = ('name', 'homepage', 'summary')
+        messages = extract_translations(upload)
+
+        for field in fields:
+            data[field] = {
+                locale: resolve_i18n_message(
```

Comment:
I don't see how that fixes it - you don't touch `locale`? Unless that happens automatically un thé translate field when you set a dict?